www/src/pages/en/usage/trpc.md

```diff
@@ -326,6 +326,40 @@ test("example router", async () => {
 });
 ```
 
+If you need to test a protected procedure then use a mock to unprotect the procedure:
+
+```ts
+import { type inferProcedureInput } from "@trpc/server";
+import { createInnerTRPCContext } from "~/server/api/trpc";
+import { appRouter, type AppRouter } from "~/server/api/root";
+import { expect, test, vi } from "vitest";
+
+test("protected example router", async () => {
+  vi.mock("../trpc", async () => {
+    const actual = await vi.importActual("../trpc");
+    
+    return {
+      ...actual,
+      protectedProcedure: actual.publicProcedure, // unprotect a protected tRPC procedure
+    };
+  });
+  
+  const ctx = await createInnerTRPCContext({ session: null });
```

Comment:
You could just mock the session:

https://github.com/juliusmarminge/t3-complete/blob/6402e59307b6e2bfca38359d20971c63c32f020d/test/trpc/post-router.test.ts#L20

Reply:
I like your solution better than mine because it doesn't require any advanced mocks. I'll update the PR with your solution.